### ISSUE TITLE
graphql-transport-ws handler doesn't free up IDs

### ISSUE THREAD

**Issue by kristjanvalur** [strawberry-graphql/strawberry#1733]
when investingating #1731 , I came across a curious thing.
A client `complete` message will cause a `self.cleanup_operation()` for the supplied ID, thus releasing server resoures and freeing it up for re-use on the same Websocket connection.
But a server-side `complete` (i.e. the subscription drains or causes an error) **will not** cause the same kind of cleanup.
From looking at the source, it would appear that Id and resources are only released at the end of the WebsocketConnection, in 
`GraphQLTransportWSHandler.handler_request`.

This is not in accordance with protocol.  the `graphql-transport-ws` spec says:
```
If there is already an **active subscriber** for an operation matching the provided ID, regardless of the operation type, the server must close the socket immediately with the event 4409: Subscriber for <unique-operation-id> already exists.
```

According to this, the ID really should be freed up as soon as the operation is done.  This would also be in accordance with good housekeeping practices, to avoid having the state machinery fill up with old subscription metadata.

**Comment by kristjanvalur** [comment]
Please see updated https://github.com/enisdenjo/graphql-ws/blob/master/PROTOCOL.md which discusses how IDs should be forgotten once complete.

**Comment by DoctorJohn** [comment]
This issue has been resolved in #1792. Thanks for the report and the PR :)
